suricata/update/main.py

```diff
@@ -1390,6 +1399,11 @@ def _main():
     # Backup the output directory.
     logger.info("Backing up current rules.")
     backup_directory = util.mktempdir()
+    if not (any(fname.endswith('.rules') for fname in os.listdir(
```

Comment:
@jasonish do you think it makes sense to consider a non-empty directory with does not have any `*.rules` file a not ideal directory for writing rules to?
Code can be improved, it'll come in the next PR after having solved conflicts if this makes sense to you.

Reply:
I think this will get tricky to maintain.  For example, at some point we'll have to move support files into the rules directory (https://redmine.openinfosecfoundation.org/issues/2688). These support files could potential have any possible filename. So that also means we may need to do conflict resolution there as well.

Adding some disucssion to:
https://redmine.openinfosecfoundation.org/issues/2691